autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/AutofillManagementRecyclerAdapter.kt

```diff
@@ -212,27 +306,43 @@ class AutofillManagementRecyclerAdapter(
     }
 
     @SuppressLint("NotifyDataSetChanged")
-    suspend fun updateLogins(
-        unsortedCredentials: List<LoginCredentials>,
-        unsortedDirectSuggestions: List<LoginCredentials>,
-        unsortedSharableSuggestions: List<LoginCredentials>,
-        allowBreakageReporting: Boolean,
+    fun showLogins(
+        autofillToggleState: AutofillToggleState,
+        promotionView: View?,
+        credentialsLoadedState: CredentialsLoadedState,
     ) {
+        Timber.v(
+            "Updating logins" +
+                ", credentialsLoadedState=$credentialsLoadedState" +
+                ", promo view: ${promotionView?.javaClass?.simpleName}",
+        )
+
         val newList = mutableListOf<ListItem>()
 
-        withContext(dispatchers.io()) {
-            val directSuggestionsListItems =
-                suggestionListBuilder.build(unsortedDirectSuggestions, unsortedSharableSuggestions, allowBreakageReporting)
-            newList.addAll(directSuggestionsListItems)
+        promotionView?.let {
+            newList.addPromotionView(promotionView)
+        }
 
-            val groupedCredentials = grouper.group(unsortedCredentials)
-            newList.addAll(groupedCredentials)
+        if (autofillToggleState.visible) {
+            newList.addTopLevelControls(autofillToggleState.enabled)
         }
 
-        withContext(dispatchers.main()) {
-            listItems = newList
-            notifyDataSetChanged()
+        when (credentialsLoadedState) {
+            is Loading -> {
```

Comment:
if it’s non-instant to retrieve the list of credentials, we don’t know yet whether to show the “empty state” buttons or not, so do nothing while it’s in `Loading` state.